File: py/plaintext.py
import framework

class Plaintext(framework.HashBenchmark):
    def __init__(self, name, *args, **kwargs):
        super().__init__(name, *args, **kwargs)
    
    # Hash a password and return the string representation
    def _hash(self, password):
        return password
    
    # Check if a hash matches a password
    def _checkHash(self, hash, password):
        return password == hash

File: cpp/plaintext.cpp
#include "framework.hpp"

class Plaintext: public HashBenchmark {
    public:
        Plaintext(std::string name) : HashBenchmark(name) {}

        std::string _hash(const std::string &password) {
            return password;
        }

        bool _checkHash(const std::string &hash, const std::string &password) {
            return password == hash;
        }
};